diff/apply/apply_linux.go

```diff
@@ -39,8 +40,11 @@ func apply(ctx context.Context, mounts []mount.Mount, r io.Reader) error {
 			}
 			return err
 		}
-		opts := []archive.ApplyOpt{
-			archive.WithConvertWhiteout(archive.OverlayConvertWhiteout),
+		opts := []archive.ApplyOpt{}
+		// OverlayConvertWhiteout (mknod c 0 0) doesn't work in userns.
+		// https://github.com/containerd/containerd/issues/3762
+		if !system.RunningInUserNS() {
```

Comment:
This will need to fall through to do the temp mount at the end of the function. Maybe just add to the case statement. Maybe do a `sync.Once`, not sure if that tradeoff is worth it.

Reply:
fixed.

`sync.Once` probably should be added to libcontainer

Reply:
I guess once the process is running that cannot change, right?

Reply:
Right. (Because multi-threaded program like Go cannot nsenter to userns)